packages/flutter/lib/src/foundation/synchronous_future.dart

```diff
@@ -42,8 +42,17 @@ class SynchronousFuture<T> implements Future<T> {
   }
 
   @override
-  Future<T> timeout(Duration timeLimit, { Future<T> onTimeout() }) => new Completer<T>().future;
+  Future<T> timeout(Duration timeLimit, { dynamic onTimeout() }) => new Completer<T>().future;
 
   @override
-  Future<T> whenComplete(Future<T> action()) => action();
+  Future<T> whenComplete(dynamic action()) {
+    try {
+      dynamic result = action();
+      if (result is Future)
```

Comment:
Yeah, I think they tried to warn on this at one point, but it causes too much noise. Especially with `Future`, it's really common to use the raw type for things like an asynchronous method that just does a side effect and doesn't return a useful value.